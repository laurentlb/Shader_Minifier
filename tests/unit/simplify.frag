#version 330

float bar(float x)
{
	float a = x;
	a = 6.;
	a = a + x;
	
	float b = x;
	b = b + x;
	b = 34.;
	b = b + x;

    float arr[2] = float[2](7.,8.);
	arr = float[2](5.,float(arr.length()));
	
	float m = 14.+length(vec3(sin(a *= 10), sin(a *= 20), sin(a *= 30)))+b++;
	m = arr[1] + arr[i++];
	m = 58.;
		
	return a + b + m;
}
float baz(float a)
{
	float b = a + 4.;
	b += sin(a);
	float c = b + 5.;
	c += sin(b);
	return c + (-(c - -c));
}

int n = 2;
float y = 47.;
out vec3 output;

float foo(float a) { if (n == 1) return 0.; if (n == 2) return a; return 1.; }
float foo(float a, float b) { if (y > a) return 0.; if (y < b) return a; return 1.; }

void notMain(float x) {
	vec3 v = vec3(foo(42.) + foo(50., 70.));
	output.rgb = v + vec3(bar(x) + baz(x));
}