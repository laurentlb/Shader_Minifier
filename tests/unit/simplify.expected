#version 330

int i;
float bar(float x)
{
  float a=6.;
  a+=x;
  float b=x;
  b+=x;
  b=34.;
  b+=x;
  float arr[2]=float[2](7.,8.);
  arr=float[2](5.,float(arr.length()));
  float m=a*=10,a*=20,a*=30,b++,i++,58.;
  return a+b+m;
}
float baz(float a)
{
  float b=a+4.;
  b+=sin(a);
  float c=b+5.;
  c+=sin(b);
  return-c-c+c;
}
out vec3 output;
void notMain(float x)
{
  output.xyz=vec3(92)+vec3(bar(x)+baz(x));
}
